Library/Homebrew/download_strategy.rb

```diff
@@ -71,25 +71,29 @@ def quiet?
   # Unlike {Resource#stage}, this does not take a block.
   #
   # @api public
-  def stage
+  def stage(&block)
     UnpackStrategy.detect(cached_location,
                           prioritise_extension: true,
                           ref_type: @ref_type, ref: @ref)
                   .extract_nestedly(basename:             basename,
                                     prioritise_extension: true,
                                     verbose:              verbose? && !quiet?)
-    chdir
+    chdir(&block)
   end
 
-  def chdir
+  def chdir(&block)
     entries = Dir["*"]
     raise "Empty archive" if entries.length.zero?
-    return if entries.length != 1
 
-    begin
-      Dir.chdir entries.first
-    rescue
-      nil
+    if entries.length != 1
+      yield if block
+      return
+    end
+
+    if File.directory? entries.first
+      Dir.chdir(entries.first, &block)
```

Comment:
I am seeing a warning at this line for the first package I install on a fresh system (https://github.com/osrf/homebrew-simulation/issues/1343). I believe the `entries` variable dispays the contents of the `HOMEBREW_CELLAR`, and for the first install, `entries.length == 1`, so the condition on line 88 is not followed

I don't know what this function is supposed to do, so I'm at a loss for suggesting a solution

Reply:
Good catch. The `chdir` stuff is mostly for resource staging - other call paths don't really care about this so I've opened #10689 to limit the occasions where we `chdir` (i.e. you have to request it by passing a block).